infra/modules/providers/azure/app-monitoring/README.md

```diff
@@ -0,0 +1,117 @@
+
+In addition to the `metric alert criteria`, this module introduces a configurable `action group` that when paired with `metric alert criteria` can be configured to trigger events like e-mail notifications.
+
+#### _More on Azure Monitoring_
+
+> "Azure Monitor maximizes the availability and performance of your applications by delivering a comprehensive solution for collecting, analyzing, and acting on telemetry from your cloud and on-premises environments.
+> All data collected by Azure Monitor fits into one of two fundamental types, metrics and logs. Metrics are numerical values that describe some aspect of a system at a particular point in time. They are lightweight and capable of supporting near real-time scenarios. Logs contain different kinds of data organized into records with different sets of properties for each type." - Source: [Azure Monitor Service Overview](https://docs.microsoft.com/en-us/azure/azure-monitor/overview)
+
+This module deploys the _**Azure Monitor**_ service in order to offer visibility into the behavior of your deployed "Azure Resources". This module is recommended for `metrics` based monitoring of any "Azure Resource" deployed alongside App Services (ex. App Service Plan, App Gateway, VM, etc.). The list of `metrics` available for configuration depends entirely on the Azure Resource chosen to be monitored. For more information on how `metrics` work, please visit the [alerts-metric-overview](https://docs.microsoft.com/en-us/azure/azure-monitor/platform/alerts-metric-overview) documentation.
+
+For monitoring of an App Service, the [app-insights](../app-insights) module is instead recommended. The [app-insights](../app-insights) module leverages an _**Azure Monitor**_ service called [application insights](https://www.terraform.io/docs/providers/azurerm/r/application_insights.html).
+
+## Module Characteristics
+
+An instance of the `app-monitoring` module deploys the _**Azure Monitor**_ service in order to provide templates with the following:
+
+- Ability to deploy Azure Monitor within a single resource group.
+
+- Ability to deploy Azure Monitor with a single set of configurable `metric alert criteria` targeting one or more Azure resources.
+
+- Ability to deploy Azure Monitor with a single set of configurable `metric alert criteria` tied to a single `action group`.
+
+Terraform resources used to define the `app-monitoring` module include the following:
+
+- [azurerm_monitor_metric_alert](https://www.terraform.io/docs/providers/azurerm/r/monitor_metric_alert.html)
+
+- [azurerm_monitor_action_group](https://www.terraform.io/docs/providers/azurerm/r/monitor_action_group.html)
+
+Cobalt modules used to deploy the `app-monitoring` module include the following:
+
+- [Service Plan Module](../service-plan)
+
+Variables used to define and deploy the `app-monitoring` module:
```

Comment:
You can provide a link to the `variables.tf` file. This way we avoid reconciling changes across the readme and the variables definition.

Reply:
Variables now live within an **Argument Reference** section.